TimberAPI/SpecificationSystem/Buildings/BuildingSpecificationDeserializer.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using Timberborn.Persistence;
+
+namespace TimberbornAPI.SpecificationSystem.Buildings
+{
+    public class BuildingSpecificationDeserializer : IObjectSerializer<BuildingSpecification>
+    {
+        private PropertyKey<string> _buildingIdKey = new PropertyKey<string>("BuildingId");
+        private PropertyKey<Building> _buildingKey = new PropertyKey<Building>("Building");
+        private ListKey<Recipe> _recipesKey = new ListKey<Recipe>("Recipes");
+        private PropertyKey<MechanicalNode> _mechanicalNodeKey = new PropertyKey<MechanicalNode>("MechanicalNode");
+
+        private readonly RecipeDeserializer _recipeDeserializer;
+        private readonly BuildingDeserializer _buildingDeserializer;
+        private readonly MechanicalNodeDeserializer _mechanicalNodeDeserializer;
+
+        public BuildingSpecificationDeserializer(
+            RecipeDeserializer recipeDeserializer,
+            BuildingDeserializer buildingDeserializer,
+            MechanicalNodeDeserializer mechanicalNodeDeserializer)
+        {
+            _recipeDeserializer = recipeDeserializer;
+            _buildingDeserializer = buildingDeserializer;
+            _mechanicalNodeDeserializer = mechanicalNodeDeserializer;
+        }
+
+        /// <summary>
+        /// This class only deserializes specification jsons, so this is not used
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="objectSaver"></param>
+        public void Serialize(BuildingSpecification value, IObjectSaver objectSaver)
+        {
+            throw new NotSupportedException();
+        }
+
+        public Obsoletable<BuildingSpecification> Deserialize(IObjectLoader objectLoader)
+        {
+            var recipes = objectLoader.Has(_recipesKey)
```

Comment:
Same here does it have to have a fallback?

Reply:
shortyl: not sure if absolutely necessary, but this did prevent some crashes